oda_api/api.py

```diff
@@ -1250,28 +1256,38 @@ def from_response_json(cls, res_json, instrument, product):
                          for d in res_json['products']['numpy_data_product_list']])
 
         if 'binary_data_product_list' in res_json['products'].keys():
-            data.extend([BinaryData().decode(d)
-                         for d in res_json['products']['binary_data_product_list']])
+            try:
+                data.extend([BinaryProduct.decode(d)
+                             for d in res_json['products']['binary_data_product_list']])
+            except:
+                data.extend([BinaryData().decode(d)
+                             for d in res_json['products']['binary_data_product_list']])
```

Comment:
why is this needed?

Reply:
The except case is just for backward compatibility, New BinaryProduct will be used in general